woocommerce/woocommerce-android
# Issue/6286 show permission dialog

Closes: #6286 


### Description


This PR introduces a new "Location permission rationale" dialog. Props to @nbradbury for suggesting this improvement :). 

This dialog is shown when the OS returns `true` for [shouldShowRequestPermissionRationale](https://href.li/?https://developer.android.com/reference/androidx/core/app/ActivityCompat#shouldShowRequestPermissionRationale(android.app.Activity,java.lang.String)). From my testing it seems the dialog is shown when the user previously declined giving the app the required permissions - if they decline multiple times the systems again stops showing the rationale. However, this behavior is not documented (AFAIK) so it might be different on different OS versions.

### Testing instructions

1. Reinstall the app
2. Log in
3. Start connecting to a reader - notice the OS location permission request is shown
4. Decline the permissions
5. Notice the "Location permission missing" dialog is shown
6. Dismiss the connection flow
7. Try to connect again and notice the new permission rationale is shown
8. Click on continue and notice the OS location permission request is shown

Try testing different scenarios related to the permissions and make sure you won't encounter any dead ends or infinite loops.

### Images/gif



https://user-images.githubusercontent.com/2261188/163759226-1870b9e9-43df-4f5f-b4a8-44d95bd4ef1d.mp4
